engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
I would be okay with this change if it does not lead to a regression. Looking at how InetSocketAddress, InetAddress and the internal InetAddressHolder work:

- InetAddress.getCanonicalHostName()  only calls InetAdress.getHostFromNameService() which will return a String representation of the IP address if it doesn't get the desired hostname from the name-service.

- InetSocketAddress.getHostName() returns the configured hostName into the InetSocketAddressHolder object (which presumably comes from the local host configuration) and if that is null then it calls InetAdress.getHostFromNameService()  (same as above)

So there is a possibility that the old call returns a better FQDN when the host is properly configured with that value, otherwise both  can return the IP address as the connect address in the worst case scenario.